x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -175,6 +221,83 @@ const getInputVarsFields = (
       } as const;
     });
 
+const getSetupFormatFromInput = (
+  input: Extract<
+    NewPackagePolicyPostureInput,
+    { type: 'cloudbeat/cis_aws' | 'cloudbeat/cis_eks' | 'cloudbeat/cis_gcp' }
+  >
+): SetupFormatGCP => {
+  const credentialsType = input.streams[0].vars?.setup_access?.value;
+  // CloudFormation is the default setup format if the integration has a CloudFormation template
```

Comment:
you are right that we should consider some code reuse, however I believe its better to do it on different PR as it might require more changes on the code on other features thats not related to this feature